flink-runtime/src/test/java/org/apache/flink/runtime/io/network/NettyShuffleEnvironmentTest.java

```diff
@@ -100,6 +105,27 @@ public void testRegisterTaskWithInsufficientBuffers() throws Exception {
 		testRegisterTaskWithLimitedBuffers(bufferCount);
 	}
 
+	@Test
+	public void testSlowIODoesNotBlockRelease() throws Exception {
+		BlockerSync sync = new BlockerSync();
```

Comment:
I guess a `OneShotLatch` would also work here if the test threads call the trigger on it.

Reply:
I used a BlockerSync since it allows blocking without throwing checked exceptions.

`OneShotLatch#await` throwing `InterruptedException` is a bit annoying. Maybe we should change that; for testing purposes it seems irrelevant?